specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json

```diff
@@ -4313,12 +4313,58 @@
         "filters": {
           "$ref": "#/definitions/BlobInventoryPolicyFilter",
           "description": "An object that defines the filter set."
+        },
+        "format": {
+          "type": "string",
+          "description": "This is a required field, it specifies the format for the inventory files.",
+          "enum": [
+            "Csv",
+            "Parquet"
+          ],
+          "x-ms-enum": {
+            "name": "format",
+            "modelAsString": true
+          }
+        },
+        "schedule": {
+          "type": "string",
+          "description": "This is a required field. This field is used to schedule an inventory formation.",
+          "enum": [
+            "Daily",
+            "Weekly"
+          ],
+          "x-ms-enum": {
+            "name": "schedule",
+            "modelAsString": true
+          }
+        },
+        "objectType": {
+          "type": "string",
+          "description": "This is a required field. This field specifies the scope of the inventory created either at the blob or container level.",
+          "enum": [
+            "Blob",
+            "Container"
+          ],
+          "x-ms-enum": {
+            "name": "objectType",
+            "modelAsString": true
+          }
+        },
+        "schemaFields": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "This is a required field. This field specifies the fields and properties of the object to be included in the inventory. Valid values for this field for the blob object type include Name, Creation-Time, Last-Modified, Content-Length, Content-MD5, BlobType, AccessTier, AccessTierChangeTime, Expiry-Time, hdi_isfolder, Owner, Group, Permissions, Acl, Snapshot, VersionId, IsCurrentVersion, Metadata, LastAccessedTime. Valid values for container object type include Name, Last-Modified, Metadata, LeaseStatus, LeaseState, LeaseDuration, PublicAccess, HasImmutabilityPolicy, HasLegalHold."
```

Comment:
"Snapshot" must and only can specify when includeSnapshots = true?
"VersionId","IsCurrentVersion" must and only can specify when  includeBlobVersions= true?

If so, I will not open "Snapshot","VersionId","IsCurrentVersion" in schemafields to customer, but add it automaticlly when user specify "includeSnapshots" or "includeBlobVersions" as true

Reply:
yes this is correct, 
Snapshot is required when includeSnapshots = true and cannot be used if includeSnapshots = false
"VersionId","IsCurrentVersion  is required when includeBlobVersions= true and cannot be used if includeBlobVersions= false